wifistats/index.js

```diff
@@ -0,0 +1,114 @@
+const exec          = require('child_process').exec;
+
+// The tools
+const airport       = require('./airport');
+const procwireless  = require('./procwireless');
+const netsh         = require('./netsh');
+
+let toolInstance;
+let supportWifiStats = true;
+
+/**
+ * Uses all available tools to query for wifi stats, and the one we found to
+ * work we store in global variable {@link toolInstance}, otherwise we set
+ * supportWifiStats to false.
+ * We call all tools in parallel and when all finishes we check the results.
+ *
+ * @return {Promise}
+ */
+function initTools() {
+    return new Promise((resolve, reject) => {
+        Promise.all([
+                getStats(airport)
+                    .then(result => resolve({
+                        tool: airport,
+                        result
+                    }))
+                    .catch(() => {}),
+                getStats(procwireless)
+                    .then(result => resolve({
+                        tool: procwireless,
+                        result
+                    }))
+                    .catch(() => {}),
+                getStats(netsh)
+                    .then(result => resolve({
+                        tool: netsh,
+                        result
+                    }))
+                    .catch(() => {})
+            ]
+        ).then(results => {
+                results.forEach(resultEntry => {
+                    if (resultEntry) {
```

Comment:
Looking at Promise.all, I think you'll get back 3 results, and each can either be an object with a tool or an empty object, and empty objects would make this truthy check pass. You might want to specifically look for something in the object as well, like resultEntry.tool.

Also the forEach will iterate through all the results. If results[0] has a tool, it'll set toolInstance and resolve. If results[2] has a tool, it'll set toolInstance again. Maybe you want to do a results.find() or some other array method or do a for-loop with an early exit.